oak_functions/loader/src/server.rs

```diff
@@ -389,65 +384,17 @@ impl wasmi::Externals for WasmState {
                 args.nth_checked(0)?,
                 args.nth_checked(1)?,
                 args.nth_checked(2)?,
                 args.nth_checked(3)?,
                 args.nth_checked(4)?,
             )),
-
-            // TODO(#2710), TODO(#2711), TODO(#2712): Remove the following code, once all extensions
-            // are called via handles.
             _ => {
-                // Careful: We assume that here for the ABI call the first two arguments are the
-                // request (which is true). We will remove this, when we call every
-                // extension through `invoke`. Here, args are off-by-one compared to
-                // `invoke_extension_with_handle` because there `args[0]` is the handle.
-                let request_ptr: AbiPointer = args.nth_checked(0)?;
-                let request_len: AbiPointerOffset = args.nth_checked(1)?;
-
-                let request = self
-                    .read_extension_args(request_ptr, request_len)
-                    .map_err(|err| {
-                        self.log_error(&format!(
-                            "Unable to read input from guest memory: {:?}",
-                            err
-                        ));
-                        OakStatus::ErrInvalidArgs
-                    });
-
-                let extension = &mut self
-                    .extensions_indices
-                    .get_mut(&index)
-                    .unwrap_or_else(|| panic!("Unimplemented function at {}", index));
-
-                let result = match request {
-                    Ok(request) => {
-                        let response = extension.invoke(request);
-                        match response {
-                            Ok(response) => {
-                                if !response.is_empty() {
-                                    // Careful: We assume every ABI call which returns a non-emtpy
-                                    // response provides these arguments
-                                    // (which is true).
-                                    let response_ptr_ptr: AbiPointer = args.nth_checked(2)?;
-                                    let response_len_ptr: AbiPointer = args.nth_checked(3)?;
-                                    self.alloc_and_write_buffer_to_wasm_memory(
-                                        response,
-                                        response_ptr_ptr,
-                                        response_len_ptr,
-                                    )
-                                } else {
-                                    Ok(())
-                                }
-                            }
-                            Err(err) => Err(err),
-                        }
-                    }
-                    Err(err) => Err(err),
-                };
-                from_oak_status_result(result)
+                // TODO(mschett): Should we use host error instead?
```

Comment:
And if so: have we used `TrapKind::Host` before?

Reply:
Or maybe this should be `TrapKind::Unreachable`? Since the actual calls should all be statically checked by wasmi at load time, right? But otherwise, `TrapKind::Host` seems sensible too.

Reply:
`Unreachable` seems good enough too, but I am not sure how to justify that I chose that.

Reply:
In their example, wasmi seems to panic with `Unimplemented function at {}` in the `_`  case [1]. Should we copy that?

 [1] https://paritytech.github.io/wasmi/wasmi/trait.Externals.html#examples